File: GeneratedEntityReaderClasses/Components/CellEaterComponent.cs
namespace GeneratedNoitaClasses;
public class CellEaterComponent : NoitaComponentBase,  iNoitaType<CellEaterComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaInt eat_probability { get; set; } = new();
	public NoitaInt ignored_material { get; set; } = new();
	public NoitaBool only_stain { get; set; } = new();
	public NoitaBool eat_dynamic_physics_bodies { get; set; } = new();
	public NoitaBool limited_materials { get; set; } = new();
	public NoitaString ignored_material_tag { get; set; } = new();
	public NoitaVector<NoitaInt> materials { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		eat_probability.Read(s);
		ignored_material.Read(s);
		only_stain.Read(s);
		eat_dynamic_physics_bodies.Read(s);
		limited_materials.Read(s);
		ignored_material_tag.Read(s);
		materials.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		eat_probability.Write(s);
		ignored_material.Write(s);
		only_stain.Write(s);
		eat_dynamic_physics_bodies.Write(s);
		limited_materials.Write(s);
		ignored_material_tag.Write(s);
		materials.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/CellEaterComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class CellEaterComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    eat_probability:NoitaInt = field(default_factory=NoitaInt)
    ignored_material:NoitaInt = field(default_factory=NoitaInt)
    only_stain:NoitaBool = field(default_factory=NoitaBool)
    eat_dynamic_physics_bodies:NoitaBool = field(default_factory=NoitaBool)
    limited_materials:NoitaBool = field(default_factory=NoitaBool)
    ignored_material_tag:NoitaString = field(default_factory=NoitaString)
    materials:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaInt))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.eat_probability.Read(s)
        self.ignored_material.Read(s)
        self.only_stain.Read(s)
        self.eat_dynamic_physics_bodies.Read(s)
        self.limited_materials.Read(s)
        self.ignored_material_tag.Read(s)
        self.materials.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.eat_probability.Write(s)
        self.ignored_material.Write(s)
        self.only_stain.Write(s)
        self.eat_dynamic_physics_bodies.Write(s)
        self.limited_materials.Write(s)
        self.ignored_material_tag.Write(s)
        self.materials.Write(s)

